src/main/scala/io/scalac/wtf/domain/User.scala

```diff
@@ -21,27 +25,50 @@ object User {
   final case object PasswordTooShort  extends ValidationError
   final case object UserAlreadyExists extends ValidationError
 
-  def validateUser(email: String, password: String): ValidatedNel[ValidationError, User] = {
-    val validation = validateEmail(email) |@| validatePassword(password)
+  private val emailPattern = "^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,6}$".r
 
-    (validation) map { case (email, password) => User(email = email, password = password)}
+  //we might need to move it to a more globally accessible place at some point  
+  implicit def dbioMonad(implicit ec: ExecutionContext) = new Monad[DBIO] {
+    def pure[A](a: A): DBIO[A] = DBIO.successful(a)
+    def flatMap[A, B](fa: DBIO[A])(f: A => DBIO[B]) = fa.flatMap(f)
   }
 
-  private val emailPattern = "^[A-Za-z0-9._%+-]+@[A-Za-z0-9.-]+\\.[A-Za-z]{2,6}$".r
+  def validateUser(email: String, password: String)(implicit ec: ExecutionContext): DBIO[Xor[NonEmptyList[ValidationError], User]] = {
+    val validatedUser = for {
+      emailV <- validateEmail(email).toValidated
+      emailNel = emailV.toValidatedNel
+      passwordV <- DBIO.successful(validatePassword(password))
+      passwordNel = passwordV.toValidatedNel
```

Comment:
Why just don't make validatePassword return a ValidatedNel? To keep it consistent with emailNel = emailV.toValidatedNel?